wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Upload.ts

```diff
@@ -0,0 +1,315 @@
+          fileIcon.classList.add("jsTooltip");
+          fileIcon.title = Language.get("wcf.global.button.delete");
+          fileIcon.addEventListener("click", (event) => {
+            const target = event.currentTarget as HTMLSpanElement;
+            (target.parentNode!.parentNode!.parentNode! as HTMLElement).remove();
+
+            EventHandler.fire("com.woltlab.wcf.media.upload", "removedErroneousUploadRow");
+          });
+
+          file.classList.add("uploadFailed");
+
+          const p = file.querySelectorAll(".columnFilename .box48 > div > p")[1] as HTMLElement;
+
+          DomUtil.innerError(
+            p,
+            Language.get(`wcf.media.upload.error.${error.errorType}`, {
+              filename: error.filename,
+            }),
+          );
+
+          p.remove();
+        }
+      } else {
+        DomTraverse.childByTag(DomTraverse.childByClass(file, "mediaInformation")!, "PROGRESS")!.remove();
+
+        if (media) {
+          const fileIcon = DomTraverse.childByTag(DomTraverse.childByClass(file, "mediaThumbnail")!, "SPAN")!;
+          this._replaceFileIcon(fileIcon, media, 144);
+
+          file.className = "jsClipboardObject mediaFile";
+          file.dataset.objectId = (media.mediaID as unknown) as string;
+
+          if (this._mediaManager) {
+            this._mediaManager.setupMediaElement(media, file);
+            this._mediaManager.addMedia(media, file as HTMLLIElement);
+          }
+        } else {
+          let error: MediaUploadError = data.returnValues.errors[internalFileId];
+          if (!error) {
+            error = {
+              errorType: "uploadFailed",
+              filename: file.dataset.filename!,
+            };
+          }
+
+          const fileIcon = DomTraverse.childByTag(DomTraverse.childByClass(file, "mediaThumbnail")!, "SPAN")!;
+          fileIcon.classList.remove("fa-spinner");
+          fileIcon.classList.add("fa-remove");
+          fileIcon.classList.add("pointer");
+
+          file.classList.add("uploadFailed");
+          file.classList.add("jsTooltip");
+          file.title = Language.get("wcf.global.button.delete");
+          file.addEventListener("click", () => file.remove());
+
+          const title = DomTraverse.childByClass(
+            DomTraverse.childByClass(file, "mediaInformation")!,
+            "mediaTitle",
+          ) as HTMLElement;
+          title.innerText = Language.get(`wcf.media.upload.error.${error.errorType}`, {
+            filename: error.filename,
+          });
+        }
+      }
+
+      DomChangeListener.trigger();
+    });
+
+    EventHandler.fire("com.woltlab.wcf.media.upload", "success", {
+      files: files,
+      isMultiFileUpload: this._multiFileUploadIds.indexOf(uploadId) !== -1,
+      media: data.returnValues.media,
+      upload: this,
+      uploadId: uploadId,
+    } as MediaUploadSuccessEventData);
+  }
+}
+
+Core.enableLegacyInheritance(MediaUpload);
+
+export default MediaUpload;
```

Comment:
For a legacy module you need to use `export = ` without `export` in front of the class for compatibility reasons.